flavors/generic/generic_flavor.go

```diff
@@ -41,12 +44,16 @@ func (f *FlavorGeneric) Inject() error {
 		return nil
 	}
 
+	f.GrpcSync.Aggregator = &f.Transports
+	f.GrpcSync.ServiceLabel = &f.ServiceLabel
+
 	f.HTTP.LogFactory = &f.Logrus
 	//TODO f.HTTP.Logger = f.Logrus.LoggerWithPrefix(f.PluginName(&f.HTTP))
 	//TODO f.HTTP.Config = f.Config.ConfigWithPrefix(f.PluginName(&f.HTTP))
 	f.LogManager.ManagedLoggers = &f.Logrus
 	f.LogManager.HTTP = &f.HTTP
 	f.StatusCheck.HTTP = &f.HTTP
+	f.StatusCheck.Transport = &f.Transports
```

Comment:
statuscheck needs transport in its own Init() and transports are available only from aggregator